client/src/pages/Feed.js

```diff
@@ -242,7 +242,12 @@ const Feed = (props) => {
   };
 
   const handleCreatePost = () => {
-    dispatchAction(TOGGLE_STATE, "createPostModal");
+    const { history, isAuthenticated } = props;
+    if (isAuthenticated) {
+      dispatchAction(TOGGLE_STATE, "createPostModal");
+    } else {
+      history.push(LOGIN);
```

Comment:
I think this replaces the #505 pull request
because this is using redirect from the code instead of a wrapper component
probably works better because this is a modal?

Reply:
Hmm, the redirect was not working for me before this change.

Reply:
I also agree with @robinv85, I think we should close #505 PR for now, since once this will be merged, it would take care of the redirecting if the user is not authenticated...

Reply:
Yeah @BMariscal I'm not saying this was working, I mean that there was probably a bug on github, I just saw someone else opened a PR too for that #668 which is the same code as here pretty much.
So I think this solution probably works better at first sight than the other in #505 so we can probably close the other ones.

Reply:
https://github.com/FightPandemics/FightPandemics/pull/668 covers when you navigate to the create-post page, this only covers when you open the create post modal.

Reply:
@silenceway Isn't #668 is exactly the same as this part? The only difference that I noticed was that the if/else are inverted. Not sure if I missed anything big, if so please let me know so I can think about that too.

Reply:
According to the design, create a post on mobile is a full page and the desktop page is modal. The design team should review that.